gdrive_sync/tasks.py

```diff
@@ -45,15 +59,131 @@ def transcode_drive_file_video(self, drive_file_id: str):
             video.save()
 
 
-@app.task
-def import_gdrive_videos():
+@app.task(bind=True)
+@transaction.atomic
+def create_resource_from_gdrive(self, drive_file_id: str):
+    """Create a WebsiteContent resource from a Google Drive file for an OCW site"""
+    drive_file = DriveFile.objects.get(file_id=drive_file_id)
+    resource = drive_file.resource
+    if not resource:
+        resource = WebsiteContent.objects.create(
+            website=drive_file.website,
+            title=drive_file.name,
+            file=drive_file.s3_key,
+            type="resource",
+            metadata={"resourcetype": get_resource_type(drive_file.s3_key)},
+        )
+    else:
+        resource.file = drive_file.s3_key
+        resource.save()
+    drive_file.resource = resource
+    drive_file.save()
+
+
+@app.task(bind=True, name="import_recent_files")
+def import_recent_files(self, last_dt: str = None, import_video: bool = False):
+    """
+    Query the Drive API for recently uploaded or modified files and process them
+    if they are in folders that match Website short_ids or names.
+    """
+    if not is_gdrive_enabled():
+        return
+    if last_dt and isinstance(last_dt, str):
+        # Dates get serialized into strings when passed to celery tasks
+        last_dt = parse(last_dt).replace(tzinfo=pytz.UTC)
+    file_token_obj, _ = DriveApiQueryTracker.objects.get_or_create(
+        api_call=DRIVE_API_FILES, for_video=import_video
+    )
+    query = (
+        f"({'' if import_video else 'not '}mimeType contains 'video/' and not trashed)"
```

Comment:
I don't think this filter is working correctly. I added a .mp4 folder to `files_final` and  it was imported and a Resource with type Video was created for it.

Reply:
good catch! Ready for another look now.